### PR TITLE
Playlist Service

### PR DESCRIPTION
**Pull request by curlydevil**
Refactored a lot SearchView and HomeView with respected controllers.
Now all views and controllers use PlaylistService to manipulate playlists and playlist items.

Also, renamed some variables to be more consistent and camelCasedCorrectly.

### CODE REVIEW

**Comment by curlydevil** [comment]
Dat would fix #45 #48 #51

File: app/search/searchController.js [hunk 1]
```diff
@@ -1,85 +1,68 @@
...
-                    for (var i = 0; i < data.items.length; i++) {
-                        vm.searchResult.push(data.items[i]);
-                        var searchItem = vm.searchResult[i];
-                        searchItem.isExisted = _.find(vm.playlist, function (item) {
-                            return item.snippet.resourceId.videoId === searchItem.id.videoId;
-                        });
+        var performSearch = function(){
+            searchResource.query({q: vm.searchText}, function (data) {
+                vm.searchResult = data.items;
+                var prms = playlist.items.length ? $q.when([]) : playlistService.fillPlaylistItems(playlist);
```
Comment (dynamokk) [inline app/search/searchController.js]: not good name of variable, or if it is necessary

File: app/search/searchController.js [hunk 2]
```diff
@@ -1,85 +1,68 @@
...
+                prms.then(function(playlistWithItems){
+                    if(!playlist.items.length)
+                        playlist = playlistWithItems;
+                    for (var i = 0; i < vm.searchResult.length; i++) {
```
Comment (dynamokk) [inline app/search/searchController.js:21]: we can use _.find or _.findindex for it

**Comment by dynamokk** [comment]
there are small comments but in general ok

**Comment by curlydevil** [comment]
Fixed remarks and #58 
Also made quick temp fix for #60

**Comment by curlydevil** [comment]
closes  #45 #48 #51 #58

**Pull request merged**